### PR TITLE
Fix wrong join in DB delete call for QueuedWebhook

### PR DESCRIPTION
**Pull request by RomainLvr**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !34177

When executing the `queuedwebhookclean` automatic action, an error occurs : 

`Uncaught Exception LogicException: Invalid JOIN type `glpi_webhooks`. in /var/www/html/glpi/src/DBmysqlIterator.php at line 660`

Patch to be applied to the King ICT TGU

### CODE REVIEW

**Comment by trasher** [comment]
LGTM, maybe a tests case is missing here.

**Review by cedric-anne — APPROVED** [review]

**Comment by trasher** [comment]
For new tests, please use directly PHPUnit (in `phpunit` dir for now).

File: phpunit/functional/QueuedWebhook.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
```
Comment (cedric-anne) [inline phpunit/functional/QueuedWebhook.php:11]:
```suggestion
```

File: phpunit/functional/QueuedWebhook.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * ---------------------------------------------------------------------
+ */
+
+namespace tests\units;
+
+use CronTask;
+use DbTestCase;
+use Webhook;
+
+class QueuedWebhook extends DbTestCase
```
Comment (cedric-anne) [inline phpunit/functional/QueuedWebhook.php:41]: It should be renamed to `QueuedWebhookTest`, and the file should be renamed too.

```suggestion
class QueuedWebhookTest extends DbTestCase
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — CHANGES_REQUESTED** [review]
Maybe you can add multiple `QueuedWebhook` enties with different `send_time` / `sent_try` and check that only the expired entries were cleaned by the `cronQueuedWebhookClean` method.

Also, you should probably call the method directly in your test, instead of testing the `CronTask::launch()` result.

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**